apps/server/src/stores/runtimeState.ts

```diff
@@ -237,35 +238,28 @@ export function loadNow(timedEvents: OntimeEvent[], eventIndex: MaybeNumber = ru
  * Loads the next event and its public counterpart
  */
 export function loadNext(
-  timedEvents: OntimeEvent[],
+  rundown: Rundown,
+  metadata: RundownMetadata,
   eventIndex: MaybeNumber = runtimeState.runtime.selectedEventIndex,
 ) {
   if (eventIndex === null) {
-    // reset the state to indicate there is no future event
-    runtimeState.eventNext = null;
+    runtimeState.eventNext = null; // reset the state to indicate there is no future event
     return;
   }
-
-  // temporarily reset this value to simplify loop logic
-  runtimeState.eventNext = null;
-
-  //TODO: do we already have a it as a list of not skipped events
-  for (let i = eventIndex + 1; i < timedEvents.length; i++) {
-    const event = timedEvents[i];
-    // we dont deal with events that are not playable
-    if (!isPlayableEvent(event)) {
-      continue;
-    }
-    runtimeState.eventNext = event;
+  const nowPlayableIndex = timedToPlayableIndex(metadata, eventIndex);
+  if (!nowPlayableIndex || nowPlayableIndex > metadata.playableEventOrder.length - 2) {
```

Comment:
question: can you add a comment explaining this conditional? Maybe it warrants a comment